### PR TITLE
[codegen/go] - Inputty Go enums

### LINKED ISSUE #7351: Generated enums in Go are not "Inputty"
Given the following schema definition for an input property

```
                "volumeType": {
                    "$ref": "#/types/x:index:EbsVolumeType",
                    "type": "string"
                },
```

And the following (shortened) definition of the referenced type:

```
        "x:index:EbsVolumeType": {
            "type": "string",
            "enum": [
                {
                    "name": "gp2",
                    "value": "gp2"
                },
                {
                    "name": "io1",
                    "value": "io1"
                },
            ]
        },
```

The Go code generator generates the following entry in to the input-typed arguments structure for the resource with the input property:

```
	VolumeType EbsVolumeType `pulumi:"volumeType"`
```

However, `EbsVolumeType` is a type alias for `pulumi.String`, which is not an `Input`.

Consequently, `Output` types cannot be supplied here, which is unexpected.

### PR DESCRIPTION
**Pull request by komalali**
# Description

Fixes #7351 

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Comment by komalali** [comment]
I have done my best to update downstream providers, but unfortunately, there are issues with [k8s](https://github.com/pulumi/pulumi-kubernetes/issues/1645) and [azure-native](https://github.com/pulumi/pulumi-azure-native/pull/974/checks?check_run_id=3011412146) that are unlikely to be resolved within the timeline needed for this PR.

I request that you review my PR based on the downstream codegen tests that do work, and the codegen tests within the pulumi/pulumi repo which test both the code generation as well as the usability of the generated enums.

File: pkg/codegen/go/gen_test.go [hunk 1]
```diff
@@ -124,21 +125,25 @@ func TestEnumUsage(t *testing.T) {
...
-				size := all[3].(*int)
-				typ := all[4].(string)
+				size := all[3].(*plant.ContainerSize)
+				typ := all[4].(tree.RubberTreeVariety)
 				assert.Equal(t, *material, "ceramic", "unexpected material on resource: %v", urn)
 				assert.Equal(t, *color, "Magenta", "unexpected color on resource: %v", urn)
-				assert.Equal(t, *size, 22, "unexpected size on resource: %v", urn)
-				assert.Equal(t, typ, "Ruby", "unexpected type on resource: %v", urn)
+				assert.Equal(t, *size, plant.ContainerSize(22), "unexpected size on resource: %v", urn)
+				assert.Equal(t, typ, tree.RubberTreeVarietyRuby, "unexpected type on resource: %v", urn)
```
Comment (komalali) [inline pkg/codegen/go/gen_test.go:146]: Note that it is no longer possible to pass in untyped strings/integers to strict enums. This is in line with how strings must be wrapped in `pulumi.String` to function correctly.

**Comment by emiliza** [comment]
@pgavlin can you take a look? 👀

File: pkg/codegen/go/doc.go [hunk 2]
```diff
@@ -88,8 +88,8 @@ func (d DocLanguageHelper) GetLanguageTypeString(pkg *schema.Package, moduleName
 		glog.Errorf("cannot calculate type string for type %q. could not find a package for module %q", t.String(), moduleName)
 		os.Exit(1)
 	}
-	if _, ok := t.(*schema.EnumType); ok {
-		return modPkg.inputType(t)
+	if typ, ok := t.(*schema.EnumType); ok {
+		return modPkg.tokenToEnum(typ.Token)
```
Comment (pgavlin) [inline pkg/codegen/go/doc.go:92]: What happens if we delete these lines entirely? I have a feeling that the call to `typeString` below will do the right thing.

**Review by pgavlin — APPROVED** [review]
Looks great! Just one question about a small simplification.

File: pkg/codegen/go/doc.go (hunk 2 above)
Comment (komalali) [reply pkg/codegen/go/doc.go]: Indeed!

**Comment by t0yv0** [comment]
Props for tests that *run* generated code! Awesome.

**Pull request merged**